tools/c7n_azure/c7n_azure/resources/apimanagement.py

```diff
@@ -45,3 +48,57 @@ class resource_type(ArmResourceManager.resource_type):
             'resourceGroup'
         )
         resource_type = 'Microsoft.ApiManagement/service'
+
+
+@ApiManagement.action_registry.register('resize')
+class Resize(AzureBaseAction):
+    """
+    Action to scale api management resource.
+    Required arguments: capacity in units and tier (Developer, Basic, Standard or Premium).
+
+    :example:
+    This policy will resize api management to Premium tier with 8 units.
+
+    policies:
+      - name: resize-api
+        resource: azure.api-management
+        filters:
+          - type: value
+            key: name
+            value: test-api
+        actions:
+          - type: resize
+            tier: Premium
+            capacity: 8
+
+    """
+
+    schema = type_schema(
+        'resize',
+        required=['capacity', 'tier'],
+        **{
+            'capacity': {'type': 'number'},
+            'tier': {'enum': ['Developer', 'Basic', 'Standard', 'Premium']}
+        })
+
+    def __init__(self, data, manager=None):
+        super(Resize, self).__init__(data, manager)
+        self.capacity = self.data['capacity']
+        self.tier = self.data['tier']
+
+    def _prepare_processing(self):
+        self.client = self.session.client('azure.mgmt.resource.ResourceManagementClient')
+
+    def _process_resource(self, resource):
+        resource['sku']['capacity'] = self.capacity

```

Comment:
Can you update the resource directly with the APIM SDK?

https://github.com/Azure/azure-sdk-for-python/blob/ab599e7c6c371543e48d210c4a11f163951d83d4/sdk/apimanagement/azure-mgmt-apimanagement/azure/mgmt/apimanagement/operations/api_management_service_operations.py#L408

Reply:
I can do that. The benefit of having a generic approach though is that it can be generalized later  and used for other resources that need scaling (if they don't need additional specific scaling parameters like size of database). Any downsizes of this generic approach?

Reply:
https://github.com/Azure/azure-sdk-for-python/issues/6335
https://github.com/cloud-custodian/cloud-custodian/issues/4376

Leaving as is for now